sql/sql_connect.cc

```diff
@@ -1000,6 +1000,28 @@ static int check_connection(THD *thd)
     bzero((char*) &net->vio->remote, sizeof(net->vio->remote));
   }
   vio_keepalive(net->vio, TRUE);
+
+  if (global_system_variables.keepalive_probes != 9)
```

Comment:
I checked manual one more time and seems like only `TCP_KEEPIDLE/TCP_KEEPALIVE` are common for win/mac/linux.
But I think having `TCP_KEEPCNT` (probes) and `TCP_KEEPINTVL` (interval) is very important, because linux defaults are too high (9 and 75).
Also I pushed code with more `#ifdef` to make it work